app/src/redux-resources/analytics/hooks/useModuleAnalytics.ts

```diff
@@ -0,0 +1,140 @@
+import { AttachedModule} from '@opentrons/api-client'
+import {
+    useTrackEvent, ANALYTICS_MODULE_COMMAND_ERROR, ANALYTICS_MODULE_COMMAND_COMPLETED
+} from '/app/redux/analytics'
+import { useModulesQuery } from '@opentrons/react-api-client'
+import { ModuleOnlyParams, NINETY_SIX_CHANNEL_WASTE_CHUTE_ADDRESSABLE_AREA, TemperatureParams } from '@opentrons/shared-data'
+import { Dict } from 'mixpanel-browser'
+import {
+    getAttachedProtocolModuleMatches,
+    getProtocolModulesInfo,
+} from '/app/transformations/analysis'
+import { useNotifyDeckConfigurationQuery } from '/app/resources/deck_configuration'
+import { useMostRecentCompletedAnalysis } from '/app/resources/runs'
+import {
+    FLEX_ROBOT_TYPE,
+    getDeckDefFromRobotType,
+    RunTimeCommand
+  } from '@opentrons/shared-data'
+
+const ANALYTIC_COMMAND_TYPES: Array<RunTimeCommand['commandType']> = [
+    'thermocycler/closeLid',
+    'thermocycler/openLid',
+    'temperatureModule/setTargetTemperature',
+    'thermocycler/setTargetLidTemperature',
+    'thermocycler/setTargetBlockTemperature',
+    'heaterShaker/setTargetTemperature',
+    'thermocycler/deactivateLid',
+    'thermocycler/deactivateBlock',
+    'temperatureModule/deactivate',
+    'magneticModule/disengage',
+    'heaterShaker/deactivateShaker',
+    'heaterShaker/deactivateHeater',
+    'heaterShaker/openLabwareLatch',
+    'heaterShaker/closeLabwareLatch' 
+];
+
+export type ModuleActionDetails = 
+    | moduleProtocolCommand
+    | moduleLiveCommand
+export interface moduleProtocolCommand {
+    runId: string,
+    action: string,
+    result: {status: string, data: any},
+    params: Dict,
+    errorDetails: string;
+}
+
+export interface moduleLiveCommand {
+    moduleType: string,
+    action: string,
+    result: {status: string, data: any},
+    serialNumber: string,
+    temperature?: number | string,
+    firmwareVersion: string
+}
```

Comment:
```suggestion

type ModuleAnalyticKind = 'protocolCommand' | 'liveCommand'

interface BaseModuleAnalytics {
kind: ModuleAnalyticKind
analyticCommand: typeof ANALYTIC_COMMAND_TYPES (this used to be "action")
result: {status: CommandStatus; data: CommandData['data'] | undefined (if needed) - do not type coerce whatever you're passing into your fn }
}

export interface ModuleAnalyticProtocolCommand extends BaseModuleAnalytics {
kind: 'protocolCommand'
[your other props]
params: CommandData['data']['params'] | undefined <- avoid using Dict unless absolutely necessary.
}

export interface ModuleAnalyticLiveCommand extends BaseModuleAnalytics {
kind: 'liveCommand'
moduleType: ModuleType
[your other props]
}
```

Let's pass the kind here so we can clearly define what interface we're working with and what sort of props are expected (this helps clarify what set of props are clearly expected from a Typescript typing perspective). We may not need the `kind` directly in the hook right now, but it may be useful in the future. 

Avoid using `any` unless this is test code.